app/Listeners/BeforeUpdateSubscriber.php

```diff
@@ -0,0 +1,121 @@
+<?php
+namespace App\Listeners;
+
+use App\Events\BeforeUpdate;
+use Illuminate\Events\Dispatcher;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Storage;
+
+class BeforeUpdateSubscriber
+{
+    public function subscribe(Dispatcher $events)
+    {
+        $events->listen(
+            BeforeUpdate::class,
+            [BeforeUpdateSubscriber::class, 'handleFileUploads']
+        );
+
+        $events->listen(
+            BeforeUpdate::class,
+            [BeforeUpdateSubscriber::class, 'hashPassword']
+        );
+    }
+
+    public function handleFileUploads(BeforeUpdate $event)
+    {
+        $data = &$event->data;
+        $model = $event->model;
+
+        $fileFields = $model::getFileFields();
+
+        foreach ($fileFields as $field => $options) {
+            if (isset($data[$field])) {
+                $event->data[$field] = $this->processField($data[$field], $options);
+            }
+        }
+        Log::info($data);
+        return $data;
+    }
```

Comment:
**Manipulação de arquivos antigos**: Se um arquivo existente no banco de dados for substituído por um novo upload, certifique-se de remover o arquivo antigo. Caso contrário, os arquivos antigos ficarão armazenados desnecessariamente, consumindo espaço.